server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -110,11 +117,13 @@ protected void asyncShardOperation(GetRequest request, ShardId shardId, ActionLi
         IndexService indexService = indicesService.indexServiceSafe(shardId.getIndex());
         IndexShard indexShard = indexService.getShard(shardId.id());
         if (indexShard.routingEntry().isPromotableToPrimary() == false) {
+            assert indexShard.indexSettings().isFastRefresh() == false
```

Comment:
Please have @pxsalehi confirm that this assertion makes sense. I believe it does, but just want to check.

Reply:
I think it makes sense. Gets on fast refresh indices should go to promotable shards. The assertion message might be a bit misleading though.

Reply:
You're right, corrected the message a bit.